### PR TITLE
godini: init at 1.0.0, treegen: init at 1.1.0

### PR DESCRIPTION
**Pull request by 4r7if3x**
## Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] `meta.mainProgram` is set, if applicable.
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden

### Comments

This is my first contribution to NixOS Packages, so I’ve added myself to the maintainers list and packaged a couple of command-line tools I developed. They both can be cross-compiled for various CPU Architectures and Operating Systems.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/go/godini/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  testers,
+}:
+
+buildGoModule rec {
+  pname = "godini";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "bilbilak";
+    repo = "${pname}";
```
Comment (pluiedev) [inline pkgs/by-name/go/godini/package.nix:16]: `repo = pname;` is an antipattern as the two are arbitrary and not necessarily equal

File: pkgs/by-name/go/godini/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    rev = "v${version}";
```
Comment (pluiedev) [inline pkgs/by-name/go/godini/package.nix:17]:
```suggestion
    tag = "v${version}";
```

File: pkgs/by-name/go/godini/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    hash = "sha256-83OAddIoJzAUXPZKGnAx8XPKrdSmtc1EIJUDmRHTU/U=";
+  };
+
+  vendorHash = "sha256-hocnLCzWN8srQcO3BMNkd2lt0m54Qe7sqAhUxVZlz1k=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/bilbilak/${pname}/config.Version=${version}"
```
Comment (pluiedev) [inline pkgs/by-name/go/godini/package.nix:26]:
```suggestion
    "-X github.com/bilbilak/godini/config.Version=${version}"
```

File: pkgs/by-name/go/godini/package.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    };
+
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    changelog = "https://github.com/bilbilak/${pname}/blob/main/CHANGELOG.md";
+    description = "INI Configuration Management Tool";
+    homepage = "https://github.com/bilbilak/${pname}";
+    license = lib.licenses.gpl3;
```
Comment (pluiedev) [inline pkgs/by-name/go/godini/package.nix:44]: gpl3Only or gpl3Plus?

File: pkgs/by-name/go/godini/package.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    mainProgram = "${pname}";
+    maintainers = with lib.maintainers; [ _4r7if3x ];
+    platforms = lib.platforms.all;
```
Comment (pluiedev) [inline pkgs/by-name/go/godini/package.nix:47]: Heavily doubt it (not even Go works on all platforms). If it works on all platforms Go supports, then just leave it blank

**Review by pluiedev — CHANGES_REQUESTED** [review]
Welcome to Nixpkgs! Couple of suggestions:

(PS: the commits have to be squashed down/split into just three: one for maintainer addition and one for each new package. The PR should also be renamed to `gobini: init at <version>, treegen: init at <version>`)

File: pkgs/by-name/go/godini/package.nix (hunk 2 above)
Comment (4r7if3x) [reply pkgs/by-name/go/godini/package.nix]: I copied that straight out of the [documentation](https://nixos.org/manual/nixpkgs/stable/#sec-language-go) and it worked too! But I understand that `tag` is more accurate and it passed my test as well.

File: pkgs/by-name/go/godini/package.nix (hunk 3 above)
Comment (4r7if3x) [reply pkgs/by-name/go/godini/package.nix]: Some other packages had used in-string variables as such, so I thought it's alright to do! For the sake of consistency, I replaced this variable with plain text everywhere...

File: pkgs/by-name/go/godini/package.nix (hunk 4 above)
Comment (4r7if3x) [reply pkgs/by-name/go/godini/package.nix]: I noticed that SPDX had set this as deprecated, but I wasn't quite convinced about using a different name for it. But never mind, I'll set it on `gpl3Only` here...

File: pkgs/by-name/go/godini/package.nix (hunk 2 above)
Comment (pluiedev) [reply pkgs/by-name/go/godini/package.nix]: Yeah, Nixpkgs's documentation is a bit dated... We really need to get on updating them as a part of sweeping treewide changes

**Comment by 4r7if3x** [comment]
@pluiedev Everything is accounted for in my last update. Thanks for the feedback!

File: pkgs/by-name/go/godini/package.nix [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  meta = {
+    changelog = "https://github.com/bilbilak/godini/blob/main/CHANGELOG.md";
+    description = "INI Configuration Management Tool";
+    homepage = "https://github.com/bilbilak/godini";
+    license = lib.licenses.gpl3Only;
+    mainProgram = "godini";
+    maintainers = with lib.maintainers; [ _4r7if3x ];
+    platforms = with lib.platforms; freebsd ++ linux ++ darwin ++ windows;
```
Comment (pluiedev) [inline pkgs/by-name/go/godini/package.nix:47]: Could condense it down

```suggestion
    platforms = with lib.platforms; unix ++ windows;
```

Comment (4r7if3x) [reply pkgs/by-name/go/godini/package.nix]: Oh, I didn't know `unix` here means Unix-like...

**Comment by 4r7if3x** [comment]
@pluiedev Everything is accounted for in my last update. Thanks for the feedback!

**Review by pluiedev — APPROVED** [review]
Diff LGTM, just have to wait for @ofborg and attention from a committer (You can get some by posting this PR to the "PRs already reviewed" thread on the NixOS Discourse)

**Review by Scrumplex — APPROVED** [review]

**Comment by 4r7if3x** [comment]
@Scrumplex Thanks for the merge! What's the process from here until the packages are listed on the Search website? I expected to find them on the unstable channel already...

**Comment by Scrumplex** [comment]
The `nixos-unstable` channel is its own branch. New changes like this always go into master (or staging, depending on the complexity) and will be merged into unstable, once Hydra CI passes.
https://github.com/NixOS/nixpkgs/blob/9640018b2247891251565852d6218d57c135a495/CONTRIBUTING.md#flow-of-merged-pull-requests

**Comment by Scrumplex** [comment]
You can follow the progress of this process for this PR here: https://nixpk.gs/pr-tracker.html?pr=368882

**Pull request merged**